### PR TITLE
Add more fail checking of data

### PR DESCRIPTION
**Pull request by Danielhiversen**
Add more fail checking of data

Tested by myself and one user here: https://community.home-assistant.io/t/xiaomi-mi-plants-monitor-flower/3388/105?u=danielhiversen

### CODE REVIEW

File: miflora/miflora_poller.py [hunk 1]
```diff
@@ -258,11 +261,16 @@ def parameter_value(self, parameter, read_cached=True):
...
             return
-        datasum = 0
-        for i in self._cache:
-            datasum += i
-        if datasum == 0:
+        if self._cache[7] > 100: # moisture over 100 procent
+            self._cache = None
+            return
+        if self._firmware_version >= "2.6.6":
+            if sum(self._cache[10:]) == 0:
```
Comment (Danielhiversen) [inline miflora/miflora_poller.py]: I am not sure about the validity of this check.

I do not no the meaning of the data from index 10. But in all the test data I have achieved the values are above 0 when the values are valid. For invalid data the values are often 0.

**Comment by Danielhiversen** [comment]
@open-homeautomation Could you have a look at this one?

**Comment by Danielhiversen** [comment]
@open-homeautomation

**Comment by open-homeautomation** [comment]
I can't say anything about the validity check with byte 10, but I recommend, we put it in and see what feedback is reported from users.

**Pull request merged**